gc/api/.clang-format

```diff
@@ -0,0 +1,114 @@
+---
+Language:        Cpp
+# BasedOnStyle:  Mozilla
```

Comment:
Was this file meant to be included?

Reply:
Yeah, this is the format I've been using for the API directory. I figured it'd be useful for anyone making a change in the code. Do you want me to drop it from the PR?